spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -145,20 +152,34 @@ protected void doStop() {
 	@Override
 	protected void onInit() {
 		Assert.notNull(this.source, "source must not be null");
-		Assert.notNull(this.outputChannel, "outputChannel must not be null");
+		Assert.state((this.outputChannelName == null && this.outputChannel != null)
+				|| (this.outputChannelName != null && this.outputChannel == null),
+				"One and only one of 'requestChannelName' or 'requestChannel' is required.");
```

Comment:
Doh. Copy/paste artifact.
Re. imports:
![optimize_imports](https://cloud.githubusercontent.com/assets/1071590/13616655/ddf9686e-e548-11e5-93a6-d2382dedfafa.jpg)

(I don't know why it doesn't work properly :cry: )

Another option is with the `Commit` function, but I don't use Git tool from IDEA for that.
Or manually with the specific short-key.

Otherwise no: we don't have any other like "On Save" options in IDEA.

So, yes, that's cool that we have Travis + Checkstyle :smile: 

Thank you if you are going to fix them all on merge!